icechunk/src/format/snapshot.rs

```diff
@@ -157,95 +150,153 @@ pub struct AttributeFileInfo {
-        total_parents: u32,
-        properties: Option<SnapshotProperties>,
+        parent_id: Option<SnapshotId>,
+        message: String,
+        metadata: Option<SnapshotProperties>,
         nodes: BTreeMap<Path, NodeSnapshot>,
         manifest_files: Vec<ManifestFileInfo>,
         attribute_files: Vec<AttributeFileInfo>,
     ) -> Self {
-        let metadata = SnapshotMetadata::default();
-        let short_term_parents = short_term_history.len() as u16;
-        let started_at = Utc::now();
-        let properties = properties.unwrap_or_default();
+        let metadata = metadata.unwrap_or_default();
+        let flushed_at = Utc::now();
         Self {
+            id: SnapshotId::random(),
+            parent_id,
+            flushed_at,
+            message,
             manifest_files: manifest_files
                 .into_iter()
                 .map(|fi| (fi.id.clone(), fi))
                 .collect(),
             attribute_files,
-            total_parents,
-            short_term_parents,
-            short_term_history,
             metadata,
-            started_at,
-            properties,
+            nodes,
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn from_fields(
+        id: SnapshotId,
+        parent_id: Option<SnapshotId>,
+        flushed_at: DateTime<Utc>,
+        message: String,
+        metadata: SnapshotProperties,
+        manifest_files: HashMap<ManifestId, ManifestFileInfo>,
+        attribute_files: Vec<AttributeFileInfo>,
+        nodes: BTreeMap<Path, NodeSnapshot>,
+    ) -> Self {
+        Self {
+            id,
+            parent_id,
+            flushed_at,
+            message,
+            metadata,
+            manifest_files,
+            attribute_files,
             nodes,
         }
     }
 
     pub fn from_iter<T: IntoIterator<Item = NodeSnapshot>>(
-        parent: &Snapshot,
+        parent_id: SnapshotId,
+        message: String,
         properties: Option<SnapshotProperties>,
         manifest_files: Vec<ManifestFileInfo>,
         attribute_files: Vec<AttributeFileInfo>,
         iter: T,
     ) -> Self {
         let nodes = iter.into_iter().map(|node| (node.path.clone(), node)).collect();
-        let mut history = parent.short_term_history.clone();
-        history.push_front(parent.metadata.clone());
 
         Self::new(
-            history,
-            parent.total_parents + 1,
+            Some(parent_id),
+            message,
             properties,
             nodes,
             manifest_files,
             attribute_files,
         )
     }
 
-    pub fn empty() -> Self {
-        let metadata =
-            SnapshotMetadata::with_message(Self::INITIAL_COMMIT_MESSAGE.to_string());
-        Self {
-            metadata,
-            ..Self::new(VecDeque::new(), 0, None, Default::default(), vec![], vec![])
-        }
+    pub fn initial() -> Self {
+        let properties = [("__root".to_string(), serde_json::Value::from(true))].into();
```

Comment:
Why not empty?

Reply:
I thought ... empty is not the main characteristic, the important part is that it is empty and has the commit message "repo initialized'

Reply:
Sorry I was talking about `properties` here, why not set it to `None`?

Reply:
I don't know... I thought we could start putting some stuff there, and reserving the `__` namespace.